openedx/core/djangoapps/content/block_structure/models.py

```diff
@@ -0,0 +1,104 @@
+"""
+Models used by the block structure framework.
+"""
+
+from django.core.files.base import ContentFile
+from django.db import models
+
+from model_utils.models import TimeStampedModel
+from openedx.core.djangoapps.xmodule_django.models import UsageKeyField
+from openedx.core.lib.block_structure.exceptions import BlockStructureNotFound
+
+
+def upload_to_path(bs_model, filename):  # pylint:disable=W0613
+    """
+    Returns path name to use for the given
+    BlockStructureModel instance.
+    """
+    return unicode(bs_model.data_usage_key)
+
+
+class BlockStructureModel(TimeStampedModel):
+    """
+    Model for storing Block Structure information.
+    """
+    class Meta(object):
+        db_table = 'block_structure'
```

Comment:
Yes, this will come out of the purge discussion.
If we make this an append-only table, we can keep track of old versions within this table and then regularly purge after some time.  If we do that, then the `data_usage_key` won't be unique.  But if we don't, then it would be.